### ISSUE TITLE
Unsure about commands

### ISSUE THREAD

**Issue by Eljensen** [evoeco/radtools#2]
I am looking for some advice on running the preprocess_ddradtags script. I have 1 lane of 2x125 bp ddRAD sequences, which are high quality. The right adapters are the standard Peterson ones (adapter, barcode (4, 5 or 6 bp long), then overhang), and the left adapters have a DBR, like this:
 
    AATT       TCAAGT   CGTCCIINNNNNN  C        AGATCGGAAGAGCACACGTCTGAACT
    Overhang   Barcode  DBR            Insert   Illumina part

We used the same left adapter for all pooled individuals, since they have unique right adapters.
 
Because I have three different length barcodes, I am running preprocess_ddradtags three times with different barcode input files. An example of my barcode file looks like this:

	4
	TCAG 1 1 NV_L26063
	AACT 1 1 NV_L26146
	AGCG 1 1 NV_L26356
	GATG 1 1 NT_100725
	GCGT 1 1 NT_100718
	ACTA 1 1 NT_100719
	TGCA 1 1 NT_100723
	CTCG 1 1 NT_100724
	CAGA 1 1 NV_X30193A
 
My command looks like this:
`$ /global/home/hpc3967/software/radtools/perl/preprocess_ddradtags.pl -l /global/home/hpc3967/polarbear/raw_ddRAD1/P_Bear_ddRAD1_R1.fastq.gz –r /global/home/hpc3967/polarbear/raw_ddRAD1/P_Bear_ddRAD1_R2.fastq.gz -b /global/home/hpc3967/polarbear/ddRAD1_demultiplexed/ddRAD_4bp_barcodes.txt -i NNNNNNMMGGACG -gz y -lt 120 -rt 120`
 
Everything runs fine, but the output doesn't find any usable reads, for example:

	Job finished in 288 min
	Total number of input reads: 227876034.
	Among them: 124557469 abb, 0 wbi, 0 noi, 11955570 woi, 28201049 wob, 63161946 dup, 0 lur, 0 rur
	 
	The counts of skipped reads are:
	CAGA: 7106963
	CTCG: 6185755
	AGCG: 1752567
	AACT: 4769132
	TCAG: 2829358
	TGCA: 7745637
	ACTA: 8973640
	GCGT: 8286859
	GATG: 15512035
	 
	Reads/loci:
	 
	barcode all_reads uniq_reads loci
	CAGA 21459938 14352975 534934
	CTCG 19743646 13557891 518069
	AGCG 7947362 6194795 396471
	AACT 14791208 10022076 452796
	TCAG 10150253 7320895 421410
	TGCA 23199475 15453838 548777
	ACTA 25458359 16484719 552802
	GCGT 24494160 16207301 522689
	GATG 40475014 24962979 612958
 
I think I may just be misunderstanding how to specify the commands or something. Any advice on how to get this working would really be appreciated.

**Comment by alephreish** [comment]
TL;DR: I believe your main problem is addressed in point 3.

1 - First of all, we have to check that the construct you have in your libraries is the same as expected by the script, because the scheme you sketched is actually for the right side and the right-side barcode is expected to be read as the index read and already parsed. So, the P7 adaptor is expected to be of this kind:

    AATT       CGTCCIINNNNNN  C       XXXXXXXXXXXXXXX   TCAAGT     XXXXXXXXXXXXXXX
    Overhang   DBR            Insert  Illumina part 1   Barcode    Illumina part 2

Such that you get the following right reads:

    G       NNNNNNMMGGACG  AATT      XXXXXXXXXXX
    Insert  DBR            Overhang  Your DNA

When you say that you used the same 'left' adaptor, do I understand you correctly that you actually mean that there was only one P7 (right) barcode?

2 - Just to be sure you create your barcode file(s) correctly. Here you list your P5 (left) barcodes with 'Ins' attached, see [here](https://github.com/evoeco/radtools/wiki/preprocess_ddradtags#barcode-files). You can put here all of your barcodes, just specify the correct number of the bases belonging to the variable part.

3 - Third, let's analyse the log you get from the script:

	124557469   abb	# too short reads
	0           wbi # OK reads
	0           noi
	11955570    woi # reads w/o DBR
	28201049    wob # reads w/o barcode
	63161946    dup # duplicates
	0           lur
	0           rur

So, **the majority of the reads actually seem to be parsed correctly**, but the read loss is due to too restrictive settings for -lt and -rt (read trimming length). This is perhaps not surprising since your 125-base long P7 reads contain 13-bases from the DBR, in addition to 1-3-base Ins -- these are removed and you are left with 109 bases. Your P5 reads are expected to contain say 6-base Ins+barcodes which are also removed, leaving you with 119 bases. So, the reads cannot be trimmed to -lt and -rt (you specify 120 for both) and are simply rejected. You have to decrease the values for -lt and -rt to some reasonable lengths - they will depend on the above calculations, on the quality of the last read bases and overall design of your experiment (the trimming is is done to ensure that your assembly program will deal with data of the same length for all of the individuals).
